### PR TITLE
Add rust-synapse-compress-state container

### LINKED ISSUE #247: Use rust-synapse-compress-state?
- Prepare docker container
- Setup deactivated in docker-compose.yaml

### PR DESCRIPTION
**Pull request by konradkonrad**
This adds build information, runner script and  service
for https://github.com/matrix-org/rust-synapse-compress-state

fixes #247

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -68,5 +68,9 @@ PFS_SERVICE_FEE=100
 # PFS_OPERATOR=<OPERATOR_NAME>
 # PFS_INFO_MESSAGE=<PFS_INFO_MESSAGE>
 
+## Settings for COMPRESS_STATE utility (optional)
+## rooms to compress: list of fully qualified room names, space separated
+# COMPRESS_ROOMS="#room1:homeserver #room2:homeserver #room3:homeserver"
```
Comment (karlb) [inline .env.template:73]: What happens when this is not set? We should either mention the default behavior or require this to be set to make it obvious.

**Review by karlb — COMMENTED** [review]
Why did you choose to use sleep instead of running a cron job?

**Comment by konradkonrad** [comment]
> Why did you choose to use sleep instead of running a cron job?

I re-used the random weekly interval from the prior `state_groups_state` cleaner tool -- I chooses a random point in the one-week future and then repeats weekly. The idea is, to avoid having all federation servers do their maintenance at the same time.
I'm also up for creating a cron-job, however, that will also need an extra installation step by RSB providers.
What do you think?

**Review by karlb — APPROVED** [review]
> I'm also up for creating a cron-job, however, that will also need an extra installation step by RSB providers.

Can't you put the cron job into the container, so that it looks the same as this approach from the outside?

>  What do you think?

It's fine either way. If this approach worked before, this is good enough for me. It just wasn't what I expected.

**Pull request merged**